src/main/java/li/zhongfu/cs2103/chatbot/types/tasks/Task.java

```diff
@@ -0,0 +1,76 @@
+package li.zhongfu.cs2103.chatbot.types.tasks;
+
+import java.io.Serializable;
+
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+/**
+ * Object that represents a to-do list task. Contains a name and a state.
+ * 
+ * Currently, it probably isn't possible for the state to be inconsistent e.g. after deserializing
+ * a (potentially malicious) serialized object, but keep this in mind if implementing subclasses
+ * with more advanced logic.
+ */
+public abstract class Task implements Serializable {
+    private String name;
+    private boolean done = false;
```

Comment:
Maybe can follow the boolean naming convention.
```suggestion
    private boolean isDone = false;
```